tests/test_api/test_validation.py

```diff
@@ -84,75 +84,215 @@ def test_check_dem_coverage():
+            },
+            {
+                'name': 'S1_136231_IW2_20200616T022313_VV_5D11-BURST'
+            }
+        ],
+        [
+            {
+                'name': 'S1_136231_IW2_20200604T022312_HH_7C85-BURST'
+            },
+            {
+                'name': 'S1_136232_IW2_20200604T022316_HH_7C85-BURST'
+            },
+            {
+                'name': 'S1_136231_IW2_20200616T022313_HH_5D11-BURST'
+            },
+            {
+                'name': 'S1_136232_IW2_20200616T022315_HH_5D11-BURST'
+            }
+        ],
+        [
+            {
+                'name': 'S1_136232_IW2_20200616T022315_VV_5D11-BURST'
+            },
+            {
+                'name': 'S1_136231_IW2_20200604T022312_VV_7C85-BURST'
+            },
+            {
+                'name': 'S1_136231_IW2_20200616T022313_VV_5D11-BURST'
+            },
+            {
+                'name': 'S1_136232_IW2_20200604T022316_VV_7C85-BURST'
+            },
+        ]
+    ]
+    invalid_cases_no_match = [
+        [
+            {
+                'name': 'S1_136231_IW2_20200604T022312_VV_7C85-BURST'
+            },
+            {
+                'name': 'S1_136232_IW2_20200616T022313_HH_5D11-BURST'
+            }
+        ],
+        [
+            {
+                'name': 'S1_136232_IW2_20200616T022315_HH_5D11-BURST'
+            },
+            {
+                'name': 'S1_136231_IW2_20200604T022312_VV_7C85-BURST'
+            },
+            {
+                'name': 'S1_136231_IW2_20200616T022313_VV_5D11-BURST'
+            },
+            {
+                'name': 'S1_136232_IW2_20200604T022316_HH_7C85-BURST'
+            },
+        ]
+    ]
+    invalid_cases_unsupported = [
+        [
+            {
+                'name': 'S1_136231_IW2_20200604T022312_VH_7C85-BURST'
+            },
+            {
+                'name': 'S1_136231_IW2_20200616T022313_VH_5D11-BURST'
+            },
+            {
+                'name': 'S1_136232_IW2_20200616T022315_VH_5D11-BURST'
+            },
+            {
+                'name': 'S1_136231_IW2_20200604T022312_VH_7C85-BURST'
+            },
+        ],
+        [
+            {
+                'name': 'S1_136232_IW2_20200616T022315_HV_5D11-BURST'
+            },
+            {
+                'name': 'S1_136231_IW2_20200604T022312_HV_7C85-BURST'
+            },
+        ]
+    ]
+    for valid_case in valid_cases:
+        validation.check_valid_polarizations({}, valid_case)
+    for (invalid_case_no_match, invalid_case_unsupported) in zip(invalid_cases_no_match, invalid_cases_unsupported):
+        with raises(validation.GranuleValidationError, match=r'.*need to have the same polarization*'):
+            validation.check_valid_polarizations({}, invalid_case_no_match)
+        with raises(validation.GranuleValidationError, match=r'.*currently supported*'):
+            validation.check_valid_polarizations({}, invalid_case_unsupported)
```

Comment:
Not a bad idea to use `zip`, but I think two separate `for` loops is more readable, like you did above.